var/spack/repos/builtin/packages/faodel/package.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Faodel(CMakePackage):
+    """Flexible, Asynchronous, Object Data-Exchange Libraries"""
+
+    homepage = "https://github.com/faodel/faodel"
+    url      = "https://github.com/faodel/faodel/archive/v1.1811.1.tar.gz"
+    git      = "https://github.com/faodel/faodel.git"
+
+    maintainers = ['fbudin69500', 'chuckatkins']
+
+    version('develop', branch='master')
+    version('1.1811.1', sha256='8e95ee99b8c136ff687eb07a2481ee04560cb1526408eb22ab56cd9c60206916')
+    version('1.1803.1', sha256='70ce7125c02601e14abe5985243d67adf677ed9e7a4dd6d3eaef8a97cf281a16')
+
+    variant('shared', default=False,
```

Comment:
Spack policy is to build shared libs by default so this should default to true

Reply:
How does Spack deal with platforms where static is preferred/required (eg. Cray CNL)?  Is the user required to set `shared=true` for each package or put it in a config file?  Or should I add something to the package file that checks for this condition?

Reply:
Spack handles the craype wrappers accordingly under the hood and sets the requisite environment variables, i.e. `CRAYPE_LINK_TYPE=dynamic`